src/components/application_manager/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -501,6 +509,241 @@ void FillUIRelatedFields(smart_objects::SmartObject& response_params,
+      patch_version;  // From generated file interfaces/generated_msg_version.h
+
+  const smart_objects::SmartObject& msg_params =
+      (*message_)[strings::msg_params];
+
+  if (msg_params[strings::language_desired].asInt() !=
+          hmi_capabilities.active_vr_language() ||
+      msg_params[strings::hmi_display_language_desired].asInt() !=
+          hmi_capabilities.active_ui_language()) {
+    LOG4CXX_WARN(logger_,
+                 "Wrong language on registering application "
+                     << application->name().c_str());
+
+    LOG4CXX_ERROR(
+        logger_,
+        "VR language desired code is "
+            << msg_params[strings::language_desired].asInt()
+            << " , active VR language code is "
+            << hmi_capabilities.active_vr_language() << ", UI language code is "
+            << msg_params[strings::hmi_display_language_desired].asInt()
+            << " , active UI language code is "
+            << hmi_capabilities.active_ui_language());
+
+    result_code = mobile_apis::Result::WRONG_LANGUAGE;
+  }
+
+  if (HmiInterfaces::STATE_NOT_AVAILABLE !=
+      application_manager_.hmi_interfaces().GetInterfaceState(
+          HmiInterfaces::HMI_INTERFACE_TTS)) {
+    FillTTSRelatedFields(response_params, hmi_capabilities);
+  }
+
+  if (HmiInterfaces::STATE_NOT_AVAILABLE !=
+      application_manager_.hmi_interfaces().GetInterfaceState(
+          HmiInterfaces::HMI_INTERFACE_VR)) {
+    FillVRRelatedFields(response_params, hmi_capabilities);
+  }
+
+  if (HmiInterfaces::STATE_NOT_AVAILABLE !=
+      application_manager_.hmi_interfaces().GetInterfaceState(
+          HmiInterfaces::HMI_INTERFACE_UI)) {
+    FillUIRelatedFields(response_params, hmi_capabilities);
+  }
+
+  if (hmi_capabilities.button_capabilities()) {
+    response_params[hmi_response::button_capabilities] =
+        *hmi_capabilities.button_capabilities();
+  }
+  if (hmi_capabilities.soft_button_capabilities()) {
+    response_params[hmi_response::soft_button_capabilities] =
+        *hmi_capabilities.soft_button_capabilities();
+  }
+  if (hmi_capabilities.preset_bank_capabilities()) {
+    response_params[hmi_response::preset_bank_capabilities] =
+        *hmi_capabilities.preset_bank_capabilities();
+  }
+  if (hmi_capabilities.hmi_zone_capabilities()) {
+    if (smart_objects::SmartType_Array ==
+        hmi_capabilities.hmi_zone_capabilities()->getType()) {
+      // hmi_capabilities json contains array and HMI response object
+      response_params[hmi_response::hmi_zone_capabilities] =
+          *hmi_capabilities.hmi_zone_capabilities();
+    } else {
+      response_params[hmi_response::hmi_zone_capabilities][0] =
+          *hmi_capabilities.hmi_zone_capabilities();
+    }
+  }
+
+  if (HmiInterfaces::STATE_NOT_AVAILABLE !=
+      application_manager_.hmi_interfaces().GetInterfaceState(
+          HmiInterfaces::HMI_INTERFACE_TTS)) {
+    FillTTSRelatedFields(response_params, hmi_capabilities);
+  }
```

Comment:
Duplicate code, same statement in line 566

Reply:
@jacobkeeler Fixed in 88fc4e0d5a15d60bd2edfc9bd623a54bd71d0243